paragraphs.type_clone.inc

```diff
@@ -0,0 +1,414 @@
+      'readonly' => 'readonly',
+      'class' => array('paragraphs-type-clone-readonly-field'),
+    ),
+  );
+  // Source paragraph type label field.
+  $form['fieldset_source']['source_label'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Label'),
+    '#default_value' => $paragraph_type->label,
+    '#required' => FALSE,
+    '#attributes' => array(
+      'readonly' => 'readonly',
+      'class' => array('paragraphs-type-clone-readonly-field'),
+    ),
+  );
+  // Source paragraph type description field.
+  $form['fieldset_source']['source_description'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Description'),
+    '#default_value' => $paragraph_type->description,
+    '#attributes' => array(
+      'readonly' => 'readonly',
+      'class' => array('paragraphs-type-clone-readonly-field'),
+    ),
+  );
+  // Source paragraph type allow_unpublish field.
+  $form['fieldset_source']['source_allow_unpublish'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Allow unpublishing from the admin interface'),
+    '#default_value' => $paragraph_type->allow_unpublish,
+    '#disabled' => TRUE,
+    '#attributes' => array(
+      'class' => array('paragraphs-type-clone-readonly-field'),
+    ),
+  );
+
+  // Target fieldset.
+  $form['fieldset_target'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Paragraph type target'),
+    '#collapsible' => FALSE,
+    '#collapsed' => FALSE,
+  );
+  // Target paragraph type name field.
+  $default_value = 'Clone of ' . $paragraph_type->name . ' (' . $num . ')';
+  $form['fieldset_target']['target_name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Name'),
+    '#default_value' => $default_value,
+    '#token' => FALSE,
+    '#required' => TRUE,
+  );
+  // Target type machine name field.
+  $form['fieldset_target']['target_machine_name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Machine name'),
+    '#default_value' => $paragraph_type_prefix . '_' . $num,
+    '#required' => TRUE,
+  );
+  // Target paragraph type label field.
+  $default_value = 'Clone of ' . $paragraph_type->label . ' (' . $num . ')';
+  $form['fieldset_target']['target_label'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Label'),
+    '#default_value' => $default_value,
+    '#required' => FALSE,
+  );
+  // Target paragraph type description field.
+  $form['fieldset_target']['target_description'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Description'),
+    '#default_value' => $paragraph_type->description,
+  );
+  // Target paragraph type allow_unpublish field.
+  $form['fieldset_target']['target_allow_unpublish'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Allow unpublishing from the admin interface'),
+    '#default_value' => $paragraph_type->allow_unpublish,
+  );
+
+  // Submit button.
+  $form['actions']['#type'] = 'actions';
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Generate'),
```

Comment:
Maybe a more descriptive action button would be better, like `Clone Paragraphs type`

Reply:
Done